### PR TITLE
NCN-3: /api/topics

### PR DESCRIPTION
**Pull request by aclearskye**
- created an app.js file to require in express and a error handling middleware function
- made a app.use route to parse req as JSON
- made a small testing route for method status to check connections to db
- made a get request to api/topics to respond with an array of items

- created tests that respond with the correct data and a 404 request.
- Implemented a error handler to catch any bad server requests

### CODE REVIEW

File: __tests__/api-topics.test.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  });
+});
+
+describe("GET:200 - /api/topics", () => {
+  it("should respond with an array of topic objects with a slug array and a description array.", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body).toEqual(topicData);
```
Comment (daviidbartlett) [inline __tests__/api-topics.test.js]: This is a very brittle test. You have no idea what order in may come back in because you haven't asserted it. You won't get punished for it on this ticket though. Future tickets you will!

File: app.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+const express = require("express");
+const {
+  getMethodStatus,
+  getTopicsData,
+} = require("./controller/topics-controller");
+const { errorMiddleware } = require("./middleware/errors");
+const app = express();
+
+app.use(express.json());
```
Comment (daviidbartlett) [inline app.js]: Why is this line here? It's not being used

File: app.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+// app.get("/api/topics", getMethodStatus);
```
Comment (daviidbartlett) [inline app.js]: remove commented code

File: controller/topics-controller.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+const {
+  articleData,
+  commentData,
+  topicData,
+  userData,
+} = require("../db/data/test-data/index.js");
+const { fetchDataFromTopics } = require("../model/topics-model.js");
+
+exports.getMethodStatus = (req, res, next) => {
```
Comment (daviidbartlett) [inline controller/topics-controller.js]: not being used and not part of this ticket

File: controller/topics-controller.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  res.sendStatus(200);
+};
+
+exports.getTopicsData = async (req, res, next) => {
+  //   fetchDataFromTopics().then((topics) => res.status(200).send(topics));
```
Comment (daviidbartlett) [inline controller/topics-controller.js]: remove commented out code

**Comment by aclearskye** [comment]
added further test assertions and passed the expected data in through to make it clearer. Removed commented code and unused, working out code
